lib/rc.js

```diff
@@ -2,11 +2,11 @@
 const fs = require('fs')
 
 function stringify (parsedData, commentChar) {
-  return Object.keys(parsedData).map(key => {
-    const v = parsedData[key]
-    const finalKey = v.comment ? `${commentChar}${key}` : key
-    return `${finalKey}=${v.value}`
-  }).join('\n') + '\n'
+  return parsedData.map(line => {
+    const finalKey = line.comment ? `${commentChar}${line.key}` : line.key
+    const separator = finalKey && line.value ? '=' : ''
```

Comment:
this doesn't see quite right - if a line is a comment, why muck with it, and assume that it has key / value parts?  In any case,  looks like a commented out setting would end up having the `=` removed from it.